### PR TITLE
Extend NameGen name lists

### PR DESCRIPTION
**Pull request by WKFO**
Mostly self-explanatory.

- The name 'Ring' is added because all orbital spaceports currently in Pioneer have at least one. If other starport types are added in future, may need to change.

- Can add more (or delete some) on request

### CODE REVIEW

File: data/libs/NameGen.lua [hunk 1]
```diff
@@ -328,6 +329,10 @@ NameGen.orbitalStarportFormats = {
 	"{name} Outpost",
 	"{name} Citadel",
 	"{name} Platform",
+	"{name} Ring",
+	"{name} Residance",
```
Comment (Web-eWorks) [inline data/libs/NameGen.lua]: Bit of a typo here...

```suggestion
	"{name} Residence",
```

Comment (WKFO) [reply data/libs/NameGen.lua]: Oops. Got confused because of my mother language :P

**Comment by robothauler** [comment]
Here are a few more orbital station names:
"{name} Trade Center",
"{name} Science Station",
"{name} Facility",
"{name} Site",
and 2, more ore less, odd city names:
"{name} Oasis",
"{name} Castle",
and for the small rocky planets e.g.
"{name} Settlement",
"{name} Commercial Area",

Please pick up if you like it.

**Comment by WKFO** [comment]
If only there was a way to assign station names according to the system's economic activities...

**Comment by impaktor** [comment]
1. To fluffy: if you merge, make sure to select "squash and merge".
2. FYI, I have a branch with _lots_ of [new names](https://github.com/impaktor/pioneer/tree/cultures/data/culture), idea was to make different regions have names from different places, so it's more like landing in a strange place where eveyrone has "pappadoppulus"-names, eg. Needs some code to assign culture mix to each system (or planet/station), if anyone wants to pick it up.
3. I assume this will be a save breaking change? If stations change name in the middle of a mission.

**Comment by WKFO** [comment]
I don't think the change affects previously generated galaxies (my old save worked) but there is a good chance that I am wrong.

**Comment by impaktor** [comment]
@WKFO the scenario I'm thinking of is: when this is merged, the stations will change name (I think), and if you have a saved mission to "Hopper High", that might now be called something else.

**Comment by WKFO** [comment]
Tested a few times. Yes, the stations do change names... However it should be relatively okay(?) since it changes name on the mission screen as well.

**Comment by impaktor** [comment]
> However it should be relatively okay(?) since it changes name on the mission screen as well

Ah, OK, then it should be fine.

**Comment by impaktor** [comment]
```
"{name} Residence",
"{name} Trade Center",
"{name} Commercial Center",
"{name} Science Station",
"{name} Observation Post",
"{name} Industrial Center",
"{name} Refinery",
```

If I'm to play devil's advocate, then these names will lead the player to believe industrial goods are cheaper / more available at a "Industrial Center", and high tech stuff at a "Science Station", etc. (Luckily, there's no `"{} Military Station"`, as that really aught to have military tech level).

With that said, I don't recommend any further action, other than squashing the commits (can be done by person who pushes green button).

**Comment by WKFO** [comment]
There must be a way to generate those station names based on system's economic activities

**Comment by impaktor** [comment]
> There must be a way (...right?) to generate those station names based on system's economic activities.

Yes, quesiton is how much C++/Lua plumming is needed. If you find the time to look into it, that would be very nice.

Thanks for the PR!

**Comment by WKFO** [comment]
It will take some free time and a smack of enthusiasm to get me going. I hate it when there is no Lua equivalent/reference of something in C++ and vice versa.

Hope this added some nice change to the game in the meanwhile.

**Pull request merged**